protocol/app/ante.go

```diff
@@ -396,6 +404,9 @@ func (h *lockingAnteHandler) otherMsgAnteHandle(ctx sdk.Context, tx sdk.Tx, simu
 		defer h.globalLock.Unlock()
 	}
 
+	if ctx, err = h.executeCosmWasm.AnteHandle(ctx, tx, simulate, noOpAnteHandle); err != nil {
```

Comment:
you should also add wasmd ante handlers :

- `CountTXDecorator`  is useful because it provides position data to the contract via [Env](https://github.com/CosmWasm/wasmd/blob/f478697fce67983546f5729b04f5389161310a8b/x/wasm/types/types.go#L271) . The motivation is to add a unique ID to the contract environment to distinguish between multiple executions within the same block but in different tx versus a reentrant call within the same msg or tx.
- `LimitSimulationGasDecorator`  limits the maximum gas available in simulations, preventing the "infinite gas" issues.
So to avoid well known issues and keep the full compatibility both ante handlers should be provided.